BugsnagPerformance/Assets/BugsnagPerformance/Scripts/Public/BugsnagPerformance.cs

```diff
@@ -332,15 +342,33 @@ private void OnRequestComplete(BugsnagUnityWebRequest request)
 
         private void EndNetworkSpan(BugsnagUnityWebRequest request, bool abort = false)
         {
-            var discard = abort || request.isHttpError || request.isNetworkError;
+            if (request == null)
+            {
+                return;
+            }
             lock (_networkSpansLock)
             {
-                if (!discard && _networkSpans.ContainsKey(request))
+                var networkSpanKey = _networkSpans.Keys.FirstOrDefault(key =>
+                {
+                    if (key.TryGetTarget(out var targetRequest))
+                    {
+                        return targetRequest == request;
+                    }
+                    return false;
+                });
+
+                if (networkSpanKey != null && _networkSpans.TryGetValue(networkSpanKey, out var span))
                 {
-                    var span = _networkSpans[request];
-                    span.EndNetworkSpan(request);
+                    if (abort || request.isHttpError || request.isNetworkError)
+                    {
+                        span.Discard();
```

Comment:
The `_onSpanEnd` callback inside `Discard()` could potentially get expensive... Can we move this logic outside of `_networkSpansLock`?

Reply:
I have changed it so that if a span is discarded then the callbacks are not run

Reply:
I Also removed the unnecessary code to remove the discarded span from the collection of potentially open spans.